aptos-move/block-executor/src/executor.rs

```diff
@@ -709,8 +705,49 @@ where
                     accumulated_fee_statement.add_fee_statement(&fee_statement);
                     counters::update_sequential_txn_gas_counters(&fee_statement);
 
+                    // TODO for materialization, we need to understand what we read,
+                    // or do exchange on every transaction, so this logic might change
+                    let mut second_phase = Vec::new();
+                    let mut updates = HashMap::new();
+                    for (id, change) in output.aggregator_v2_change_set().into_iter() {
+                        match change {
+                            AggregatorChange::Create(value) => {
+                                updates.insert(id, value);
+                            },
+                            AggregatorChange::Apply(apply) => {
+                                match apply.get_apply_base_id(&id) {
+                                    ApplyBase::Previous(base_id) => {
+                                        updates.insert(
+                                            id,
+                                            apply
+                                                .apply_to_base(
+                                                    data_map.fetch_aggregator(&base_id).unwrap(),
+                                                )
+                                                .unwrap(),
+                                        );
+                                    },
+                                    ApplyBase::Current(base_id) => {
+                                        second_phase.push((id, base_id, apply));
+                                    },
+                                };
+                            },
+                        }
+                    }
+                    for (id, value) in updates.into_iter() {
+                        data_map.write_aggregator(id, value);
+                    }
+                    for (id, base_id, apply) in second_phase.into_iter() {
+                        data_map.write_aggregator(
+                            id,
+                            apply
+                                .apply_to_base(data_map.fetch_aggregator(&base_id).unwrap())
+                                .unwrap(),
```

Comment:
Same here? Should not we be doing this in `execute_transaction` where we have `materialize_deltas` hook for sequential execution?

Reply:
hmmm. Didn't notice that

what is the reason for such a design - for having a code specially for sequential execution guarded by a flag and done in the common path, instead of just doing it in the sequential execution code?

it seems to me that this function is equivalent to execute(), which is called in parallel execution and after execute_transaction finishes does all the work of updating MVHashMap.

this here is the work of updating Unsync map, so seems to correspond here based on that?

Reply:
cc @gelash